### ISSUE TITLE
Race inheritance issue

### ISSUE THREAD

**Issue by ajjr1996** [JonStryker/Witcher#140]
The whole race inheritance is a mess. I'm going to try to think of a way to basically revamp the system to make sense because as of right now it seems like they'll inherit halfdwarf for example and stick with that through the generations. We need it to flow smoothly from full dwarf, half dwarf, and back to human or dwarf same for all races (maybe throw in 1/4 variation of their race, such as quarter elf quadroons?). But we also have think there's also for example halfelf/halfhuman nd halfelf/halfhalfling for example. The whole thing gets sorta confusing when you take in to account all races, but as of right now it's completely messy. I'm sure we could figure out a way to represent the races better it just takes some thinking.

**Comment by ajjr1996** [comment]
At least we need to figure out a way to keep it under control and not have like for instance the Aedirnian royal family marry a dwarf and have half dwarf children and remain half dwarfs even though they've bred with humans for generations. This will also cause them to sometimes fully switch to the elder race culture and religion as well. Just let me know any ideas, I'll need to discuss this a bit because I'm kind of not sure what I should do yet and would like some input.

**Comment by ajjr1996** [comment]
Also, if we want to make it simple we could maybe just disable marriage/procreation between non-humans for now. The current system works between humans and non-humans but is flawed when it comes to the interracial non-human unions (dwarf/elf gnome/dwarf, etc).

**Comment by rquinio** [comment]
Note: there is no notion of race in vanilla, so the mod uses religion groups + traits to emulate this.

Marriages between non-human races are already disabled (that's now the default in vanilla: unless explicit intermarry is allowed, marriage between religion groups is blocked with "Will not consider offers from infidels" cf https://ck2.paradoxwikis.com/Diplomatic_actions#Arrange_marriage)

So I guess the problem is that some non-humans are changing their religion.
The "demand religious conversion" diplomatic decision is already disabled, so possibly the same must be done with other vanilla decisions/events that affect religion.

**Comment by ajjr1996** [comment]
I actually have gone about re-writing the racial traits. I'm not sure how it will work out, but, here's a draft of the W_racial_events.txt to show you my idea. Still have tons of work to do but I thought it would actually work a lot better and make things a bit more organized by basically making the racial traits specifically apply to that race, for instance "halfelf_race_gnome" (gnome that's half elven) or "halfdwarf_race_human" (human that's half dwarf. Give it a look! 

[W_racial_events.txt](https://github.com/JonStryker/Witcher/files/4017700/W_racial_events.txt)

**Comment by rquinio** [comment]
Can gnome/halfling/dwarves/elves actually interbreed in the lore ?
I imagined this was limited to humans, since others were not mentioned.

**Comment by ajjr1996** [comment]
I don't think they mentioned it, I was just bored honestly and wanted to see what I could do, I'll finish my idea and show you and if you don't like it we can revert back. I just think it'd be cool to either implement this or just make it so elder races can't interbreed in-game except with humans in certain cases. It would simplify it a bit to make it so it stays a bit more organized. The race traits get a bit chaotic late game.

**Comment by rquinio** [comment]
I'd go for option 2), especially as CK2 characters represent the nobility, I guess half races wouldn't be accepted as legitimate heirs in most cases.
You could do a pull-request (in a separate branch than master) to get feedback on option 1), but IMO it seems overly complicated (what about cases like 1/4 elf, 1/4 dwarf, 1/2 human, etc.)

**Comment by ajjr1996** [comment]
Yeah, it's cool and all but not really practical. The non-humans should completely be barred from marrying outside their race. Also, I was thinking maybe tolerant humans could marry non-humans but that's the only acception.

**Comment by ajjr1996** [comment]
When two half elves breed it equals a fullblood elf with the current setup

**Comment by ajjr1996** [comment]
Maybe we could use a system where the father's race is the one inherited by default or the mother's if marriage is matrilineal. Then we could add a trait named "elf blood" or "human blood" etc. as an extra trait applies to the mother's base race trait if she is of another race. For example if the character's parents are - father: 1/2 elf (base_race) 1/2 human (human blood) and mother: 1/2 dwarf (base race) 1/2 gnome (gnome blood) the child would would be an elf (base race) with "dwarf blood" as a side trait. We wouldn't inherit old "human blood" and old "gnome blood" from the child's parents and basically update it with the new "****blood" based upon the base races. This would keep all characters clean AND be quite pleasant gameplay-wise.

**Comment by ajjr1996** [comment]
Didn't do a ton of research on how possible this method is yet, I'm just brainstorming.

**Comment by ajjr1996** [comment]
I wonder why so many characters that become halfelf or halfdwarf etc seem to commonly switch from their human culture/religion to the non-human culture/religion. I've found several cases everytime I run a test. It's especially annoying because the King of say Kaedwen in this example below the king becomes dwarven culture/religion (I'm not sure how, maybe from the dwarven parent being the guardian??? Or secret event I don't know about?) and has religious revocation on and basically wiped out all the other lords to add dwarven ones.

**Comment by ajjr1996** [comment]
![20200105164640_1](https://user-images.githubusercontent.com/16552034/71787428-39aae380-2fdd-11ea-87ee-89099ef37c89.jpg)

**Comment by ajjr1996** [comment]
![20200105164644_1](https://user-images.githubusercontent.com/16552034/71787432-40395b00-2fdd-11ea-8e3c-fa52bf2b707d.jpg)

**Comment by ajjr1996** [comment]
![20200105192141_1](https://user-images.githubusercontent.com/16552034/71789437-b1ced480-2ff0-11ea-8df8-e8b00a7d1ad3.jpg)

**Comment by ajjr1996** [comment]
![20200105192138_1](https://user-images.githubusercontent.com/16552034/71789443-b7c4b580-2ff0-11ea-9067-a0a227eae773.jpg)
